appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/palette/YoungAndroidPalettePanel.java

```diff
@@ -330,12 +464,14 @@ private static boolean showCategory(ComponentCategory category) {
   public void loadComponents(DropTargetProvider dropTargetProvider) {
     this.dropTargetProvider = dropTargetProvider;
     for (String component : COMPONENT_DATABASE.getComponentNames()) {
+      if(filter.shouldShowComponent(component) && osFilter.shouldShowComponent(component))
       this.addComponent(component);
```

Comment:
Code style: Space between if and condition, plus make sure to use curly braces around the if body.